--- a/cmake/onnxruntime.cmake
+++ b/cmake/onnxruntime.cmake
@@ -256,8 +256,15 @@ target_link_libraries(onnxruntime PRIVATE
     ${onnxruntime_EXTERNAL_LIBRARIES}
 )
 
+option(ORT_DLL_SOVERSION "DLL SOVERSION" "")
 if(WIN32)
   target_link_options(onnxruntime PRIVATE ${onnxruntime_DELAYLOAD_FLAGS})
+  if(ORT_DLL_SOVERSION)
+    set_target_properties(onnxruntime PROPERTIES
+      SOVERSION "${ORT_DLL_SOVERSION}"
+      DLL_NAME_WITH_SOVERSION ON
+    )
+  endif()
 endif()
 #See: https://cmake.org/cmake/help/latest/prop_tgt/SOVERSION.html
 if(NOT APPLE AND NOT WIN32)
